vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -559,6 +560,38 @@ def row_parallel_weight_loader(param: torch.Tensor,
     return default_weight_loader(param, loaded_weight)
 
 
+LoaderFunction = Callable[[torch.Tensor, torch.Tensor], torch.Tensor]
+
```

Comment:
The `weight_loader` is function typically implemented by the layer like `ColumnParallelLinear`. I see that these are used by both Jamba and Mamba for loading weights that are not part of a specific "layer", so having these shared functions makes sense.

That being said, its a bit confusing since this logic is effectively duplicated in the weight loader of `ColumnParallelLinear`. But the names these functions are generic. So we might consider keeping these local to the mamba and jamba files for now.

I think this is fine if you disagree, just wanted to give my gut reaction

Reply:
I am just adding a couple of generic utilities here to compliment the `default_weight_loader` and `row_parallel_weight_loader` that are already here. I'm pretty sure we'll find them useful for other models as they are highly generic.

`sharded_weight_loader` gives you a weight loader that operates on a tensor of any rank along any axis and is much simpler than the one in `ColumnParallelLinear`. Note that we're using it to shard a vector along axis 0 and `ColumnParallelLinear` shards matrices along axis 1.

`composed_weight_loader` gives us a handy way to apply any function to a tensor after loading it with any weight loader.